compat/unistd.cc

```diff
@@ -23,18 +23,34 @@
 #define _S_IWRITE 0x0080
 #endif
 
+/// true if handle is a socket, false if a file or error
+static bool
+isSocket(intptr_t handle)
+{
+    if (!isValidSocketHandle(handle)) {
+        // errno is already set by _get_osfhandle()
+        return false;
+    }
+
+    int l_so_type = 0;
+    int l_so_type_siz = sizeof(l_so_type);
+    // use the Windows API directly
+    if (getsockopt(handle, SOL_SOCKET, SO_TYPE, reinterpret_cast<char *>(&l_so_type), &l_so_type_siz) == 0)
```

Comment:
To use camelCase and avoid odd abbreviations:

```suggestion
    // we only use getsockopt() return value, but its API requires us to supply these
    int value = 0;
    int valueSize = sizeof(l_so_type);
    // use Windows API directly
    if (getsockopt(handle, SOL_SOCKET, SO_TYPE, reinterpret_cast<char *>(&value), &valueSize) == 0)
```